Core/include/Acts/EventData/SpacePointContainer.ipp

```diff
@@ -0,0 +1,271 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <cmath>
+
+namespace Acts {
+
+template <typename container_t, template <typename> class holder_t>
+template <template <typename> class, typename>
+SpacePointContainer<container_t, holder_t>::SpacePointContainer(
+    const Acts::SpacePointContainerConfig& config,
+    const Acts::SpacePointContainerOptions& options, container_t& container)
+    : m_config(config.toInternalUnits()),
+      m_options(options.toInternalUnits()),
+      m_container(container) {
+  this->initialize();
+}
+
+template <typename container_t, template <typename> class holder_t>
+template <template <typename> class, typename>
+SpacePointContainer<container_t, holder_t>::SpacePointContainer(
+    const Acts::SpacePointContainerConfig& config,
+    const Acts::SpacePointContainerOptions& options, container_t&& container)
+    : m_config(config.toInternalUnits()),
+      m_options(options.toInternalUnits()),
+      m_container(std::move(container)) {
+  this->initialize();
+}
+
+template <typename container_t, template <typename> class holder_t>
+void SpacePointContainer<container_t, holder_t>::initialize() {
+  m_data.resize(this->size(), m_config.useDetailedDoubleMeasurementInfo);
+  m_proxies.reserve(this->size());
+  for (std::size_t i(0); i < this->size(); ++i) {
+    m_data.setX(i, this->container().x_impl(i) - m_options.beamPos[0]);
+    m_data.setY(i, this->container().y_impl(i) - m_options.beamPos[1]);
+    m_data.setZ(i, this->container().z_impl(i));
+    m_data.setRadius(
+        i, std::sqrt(m_data.x(i) * m_data.x(i) + m_data.y(i) * m_data.y(i)));
+    m_data.setPhi(i, atan2f(m_data.y(i), m_data.x(i)));
```

Comment:
std::atan2 ?

Reply:
I don't know why but I get a `'atan2f' is not a member of 'std'; did you mean 'atan2'?` despite I include cmath or math.h ... don't know why

Reply:
The docu is suggestive that just atan2 should be equal good here and in case one day double is used for positions the code will adapt.